src/main/java/org/semux/consensus/SemuxBft.java

```diff
@@ -995,12 +994,15 @@ protected void applyBlock(Block block) {
         WriteLock lock = kernel.getStateLock().writeLock();
         lock.lock();
         try {
-            // [7] flush state to disk
+            // [7] add block to chain
+            chain.addBlock(block);
+
+            // [8] commit state updates
             chain.getAccountState().commit();
             chain.getDelegateState().commit();
 
-            // [8] add block to chain
-            chain.addBlock(block);
+            // [9] commit pending blockchain updates
+            chain.commit();
```

Comment:
Should chain.commit also commit the delegate/account state?  Might ensure better consistency